lib/store.js

```diff
@@ -1,24 +1,38 @@
+import { lru } from 'tiny-lru'
+import { DEFAULT_OPTIONS } from './constants.js'
 export class Store {
-  constructor(timeWindowMs) {
+  constructor(
+    timeWindowMs,
+    intervalTimeExpiredKeysEvictionMs,
+    cacheSize = DEFAULT_OPTIONS.cacheSize
+  ) {
     this.timeWindowMs = timeWindowMs
-    this.resetStore()
-    this.interval = setInterval(() => this.resetStore(), timeWindowMs)
-  }
-
-  resetStore() {
-    this.hits = {}
+    this.cache = lru(cacheSize, timeWindowMs)
+    this.interval = setInterval(() => {
+      this.cache.keys().forEach(key => {
```

Comment:
Seeing we've now switched to using tiny-lru, I don't quite understand why we're doing this then. why don't we just leave to the cache the responsibility to remove entries based on its internal logic?

Reply:
but that's the job of the cache library we use to take care of that, so I think this is an unnecessary optimization we're doing, which complicates the implementation without good reason, while also introducing a new fairly obscure option. let's remove this and the option please